### ISSUE TITLE
Gulp tasks can't complete using Gulp 4.0.2 and Node 12.2.0

### ISSUE THREAD

**Issue by nathanp** [understrap/understrap-child#229]
I'm on Windows 10 and updated Node from 11.x to 12.2.0 and now none of my gulp tasks are working.

I've tried:
- Updating the gulpfile.js and package.json
- Removing the `node_modules` folder and running `npm install` again.

Any ideas?

**Comment by xicubed** [comment]
Same on mac. I got understrap in its own folder to run npm install but I had to add "fsevents": "^2.0.6", in the package.json and I think had to pry into the node_modules and make sure node_sass was at 4.12 instead of 4.11. But now if I include understrap 0.9.3 in my child theme gulp file it fails on npm install. Do I need it? idk probably but how can I modify it in the same way if it's just a requirement that wont install/create a folder in node_modules?

if I npm install understrap (0.9.3) I get and error about file not found pointing to ......./wp-content/themes/core-understrap-1-0-7/node_modules/understrap/node_modules/undescores-for-npm/js/skip-link-focus-fix.js which is an invalid directory shouldn't have two node_modules...

**Comment by ken-mills** [comment]
Getting same error on child npm install. Any way suggestions for working through this issue?

```
[08:58:35] Finished 'copy-assets' after 17 ms
events.js:174
      throw er; // Unhandled 'error' event
      ^
Error: File not found with singular glob: ......wp-content/themes/understrap-child/node_modules/understrap/node_modules/undescores-for-npm/js/skip-link-focus-fix.js (if this was purposeful, use `allowEmpty` option)
    at Glob.<anonymous> (.....\wp-content\themes\understrap-child\node_modules\glob-stream\readable.js:84:17)
```

**Comment by arslanfarooq** [comment]
What a time waster this theme is.

**Comment by monkishtypist** [comment]
I had been scratching my head on this all day. Here is my "fix"...

TL;DR: remove `understrap` dependency from child theme `package.json`

The problem is the directory/file does not exist when running the `copy-assets` script contained in the `understrap` dependency:
```
> understrap@0.9.3 postinstall /var/www/understrap-child-demo/wp-content/themes/understrap-child/node_modules/understrap 
> gulp copy-assets

[08:09:10] Using gulpfile /var/www/understrap-child-demo/wp-content/themes/understrap-child/node_modules/understrap/gulpfile.js 
[08:09:11] Starting 'copy-assets'...
[08:09:12] Finished 'copy-assets' after 47 ms 
events.js:174 
      throw er; // Unhandled 'error' event
      ^

Error: File not found with singular glob: /var/www/understrap-child-demo/wp-content/themes/understrap-child/node_modules/understrap/node_modules/undescores-for-npm/js/skip-link-focus-fix.js (if this was purposeful, use `allowEmpty` option)
```
Note in the above error message that (1) it is using the `gulpfile` from the `understrap` dependency, and (2) this is causing `File not found` error - see the duplicated `node_modules` folders in the glob. 

This is caused by the `understrap` dependency in the child theme `package.json`.

Here's what's happening:
- When installing or updating the child theme with npm, the `understrap` dependency has its own `package.json` file containing the script `"postinstall": "gulp copy-assets"`
- That script uses a variable for the environment path (taken from `node_modules/understrap/gulpconfig.json`
- However, since it is being called as a dependency from within the understrap child theme, the `paths.node` variable gets set to the current working directory `/understrap-child/node_modules/understrap/` and appends `./node_modules/`
- Thus `paths.node` now equals `/understrap-child/node_modules/understrap/node_modules/` which does not exist
- And this causes the `understrap` dependency script `"postinstall": "gulp copy-assets"` to fail since no such directory exists.

I was able to solve this by removing `understrap` from the `package.json` dependencies in the child theme. And since this is a child theme and must be installed alongside the parent theme, I simply updated the references in `gulpfile.js` to point to `../understrap` instead of the non-existent node_modules directory.

**Comment by urasone** [comment]
Could you please show us which part of the code  was updated with `../understrap`? I cannot see node_modules there in the gulpfile.js.

Did you mean to edit  `"node": "./node_modules/",` to   `"node": "../understrap",` within gulpconfig.json instead of gulpfile.js?

I could find there:
```
  "paths" : {
    "js": "./js",
    "css": "./css",
    "img": "./img",
		"imgsrc": "./src/img",
    "sass": "./sass",
    "node": "./node_modules/",
    "bower": "./bower_components/",
    "dev": "./src",
    "dist": "./dist",
    "distprod": "./dist-product",
    "vendor": ""
  }

```

**Comment by monkishtypist** [comment]
Basically delete line 52 from `package.json` in the child theme:
https://github.com/understrap/understrap-child/blob/8855739aec4dfa9767f03ba692e55c774bc061b8/package.json#L52

Then comment out lines 197 to 199 in `gulpfile.js` in the child theme:
https://github.com/understrap/understrap-child/blob/8855739aec4dfa9767f03ba692e55c774bc061b8/gulpfile.js#L197-L199

If you need the sass files from the understrap parent theme, they are already imported from the `src` directory of the child theme.

However, the child theme `src` directory won't update if you update the parent `understrap` theme, so alternately you could replace the following line:
https://github.com/understrap/understrap-child/blob/8855739aec4dfa9767f03ba692e55c774bc061b8/sass/child-theme.scss#L3
with:
```scss
@import "../../understrap/sass/theme"; // <------- Load understrap parent theme sass files
```

Play around with it and you should figure it out from there.

**Comment by raisonon** [comment]
Just to confirm @monkishtypist suggestion worked for me. Thanks!

https://github.com/understrap/understrap-child/issues/229#issuecomment-554764600

**Comment by Thomas-A-Reinert** [comment]
Yes, that worked for me too. Thanks!

**Comment by bacoords** [comment]
The npm/gulp tasks have been addressed in the latest update. If new errors have persisted, please open a new issue so we can address them.
